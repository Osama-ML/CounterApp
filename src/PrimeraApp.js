import React from "react";
import PropTypes from "prop-types";

const PrimeraApp = ({ saludo, subtitulo }) => {
  return (
    <>
      <h1>Bienvenido ยก{saludo}!</h1>
      {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> */}
      <p>{subtitulo}</p>
    </>
  );
};

PrimeraApp.protoTypes = {
  saludo: PropTypes.string.isRequired,
};

PrimeraApp.defaultProps = {
  saludo: "Osama",
  subtitulo: "Soy un subtitulo",
};

export default PrimeraApp;
