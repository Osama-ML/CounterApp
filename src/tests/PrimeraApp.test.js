import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import React from "react";
import { shallow } from "enzyme";

describe("Prueba en PrimeraApp", () => {
  // test('debe mostrar el mensaje "Bienvenido ¡Osama!"', () => {
  //   const saludo = "Bienvenido ¡Osama!";

  //   const wrapper = render(<PrimeraApp saludo="Osama" />);

  //   expect(wrapper.getByText(saludo)).toBeInTheDocument();
  // });
  test("debe de mostrar <PrimeraApp/> correctamente", () => {
    const wrapper = shallow(<PrimeraApp saludo="Osama" />);
    const saludo = "Bienvenido ¡Osama!";
    expect(wrapper).toMatchSnapshot(saludo);
  });

  test("debe mostrar los subtitulos enviados en las propiedades", () => {
    const saludo = "Bienvenido ¡Osama!";
    const subtitulo = "Soy un subtitulo de testing";
    const wrapper = shallow(
      <PrimeraApp subtitulo={subtitulo} saludo="Osama" />
    );
    const textoParrafo = wrapper.find("p").text();
    expect(textoParrafo).toBe(subtitulo);
  });
});
