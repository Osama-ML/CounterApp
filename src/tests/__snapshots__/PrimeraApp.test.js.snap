// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en PrimeraApp debe de mostrar <PrimeraApp/> correctamente: Bienvenido ¡Osama! 1`] = `
<Fragment>
  <h1>
    Bienvenido ¡
    Osama
    !
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
